pool/paymentmgr.go

```diff
@@ -927,11 +961,13 @@ txConfs:
 			// Non-blocking receive fallthrough.
 		}
 
-		resp, err := notifSource()
+		resp, err := fetchTxConfNotifications(ctx, notifSource)
 		if err != nil {
-			desc := fmt.Sprintf("%s: unable to fetch tx confirmations: %v",
-				funcName, err)
-			return poolError(ErrTxConf, desc)
+			if errors.Is(err, ErrContextCancelled) {
```

Comment:
Cancelled context is being handled here, so I think the `select` statement above is now redundant and can be removed.

Reply:
good catch, simplifying that.